server/src/house/house.service.ts

```diff
@@ -5,10 +5,32 @@ import { PrismaService } from '../prisma/prisma.service';
 export class HouseService {
   constructor(private prisma: PrismaService) {}
 
+  async getHouse(houseId: number) {
+    try {
+      const house = await this.prisma.house.findUnique({
+        where: { id: houseId },
+        include: {
+          rules: true,
+          userHouses: true,
+          tasks: true,
+          expenses: true,
+        },
+      });
```

Comment:
This getHouse API does not have to return expenses and tasks because I think we will make another API for them. So could you delete them?
And if you return the array of usreHouse, you have to get the house in client side.  So I think the following is better way! You can refer to GetMe!

API need to return these data

```js
      const houseResponse = {
        id: house.id,
        name: house.name,
        isExpensePerTime: house.isExpensePerTime,
        rules: house.rules,
        isUserAdmin: house.userHouses[0].isAdmin, // Not sure if we need, but just in case
      };
```

Also you need to filter not only `houseId` but also using `userId` too!